src/ast/CMakeLists.txt

```diff
@@ -39,9 +38,28 @@ if (STATIC_LINKING)
       clangSema
       clangSerialization
       clangToolingCore)
-  llvm_map_components_to_libnames(llvm_libs bpfcodegen ipo irreader mcjit orcjit ${LLVM_TARGETS_TO_BUILD})
-  target_link_libraries(ast ${clang_libs})
-  target_link_libraries(ast ${llvm_libs})
+
+  if(EMBED_CLANG)
+    target_link_libraries(ast ${CLANG_EMBEDDED_CMAKE_TARGETS})
+    list(APPEND clang_libs clangToolingInclusions)
+    if(EMBED_LIBCLANG_ONLY)
+      foreach(clang_lib IN LISTS clang_libs)
+        get_target_property(link_libraries ${clang_lib} INTERFACE_LINK_LIBRARIES)
+        string(REPLACE "LLVM" "" static_only_libs "${link_libraries}")
```

Comment:
Wanted to flag this as something to revisit in #1063 -- this could be fragile if clang ever links against another lib whose name contains "LLVM" as a substring. I ran into this on the other PR.

Reply:
@acj hope you don't mind, I copied the function from your branch into embed_helpers.cmake with a couple of fixes. It was similar to work I had from another branch so I figured it was fine, and this way you can just use it when you rebase. It fixed my linker issue anyways!

Reply:
I don't mind at all. Glad to hear that it was useful :)